main/src/cgeo/geocaching/filters/core/DifficultyAndTerrainGeocacheFilter.java

```diff
@@ -0,0 +1,64 @@
+package cgeo.geocaching.filters.core;
+
+import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.storage.SqlBuilder;
+import cgeo.geocaching.utils.expressions.ExpressionConfig;
+
+public class DifficultyAndTerrainGeocacheFilter extends BaseGeocacheFilter {
```

Comment:
It is an interesting idea that your  new filter reuses the two existing ones. I rather thought about creating a new filter andthen delete the two other ones (as they are no longer necessary)

Reply:
It just came into my mind - and I thought just give it a try 😇 . If it does not work, I can still can create a independent one